### DISCUSSION TITLE
Is downloading all songs (video) possible?

### DISCUSSION THREAD

**Discussion by ux-git** [sircharlo/meeting-media-manager#2196] [Ideas] [answered]
Is it possible to download all the song media files to local storage for usage either without Internet or while facing slow/unstable connection? Sometimes our connection is like only a few Mbps, taking tens of minutes to sync all the media (still surprised how Zoom manage to work under these conditions). In this case syncing images is quite acceptable since their size is far smaller than videos.

Is there's no way currently, this might be a "nice-to-have" feature request 🙂

**Comment by e-Feitosa** [comment]
You can practically download the entire month's meeting, look at the software's home screen, change the week and click on download media. This way, you can keep the downloaded media available for all weeks.
⬇️
![image](https://github.com/sircharlo/meeting-media-manager/assets/90984283/2c1d6c23-253e-45b2-a92a-58e833504b26)

**Reply by ux-git** [reply]
Thanks, sounds like a good temp/patch solution. Unfortunately this won't fix "Last minute song" to be available offline. But that's what I'll be using for sure until the feature is implemented 🤝

**Comment by sircharlo** [comment, marked as answer]
I see the merits for this. @mtdvlpr do you think we could add this to the refactor roadmap? Perhaps under "advanced", we could have a cache section where one could cache various things: MP3 songs for background music; all MP4 song videos. Also, the controls to clear the cache would be there as well. What do you think?

**Reply by mtdvlpr** [reply]
Completely agree!

**Reply by ux-git** [reply]
Yes, that would be awesome! Offline playback of MP3 songs is great feature and it works, so having the same option for MP4 song videos will be so useful! Thank you both!
